client/redux/features/user/slice.ts

```diff
@@ -0,0 +1,21 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { UserData } from "@/redux/index";
+
+// 🔴 State: set initial state of slice
+const initialState = {
+  data: [],
+} as UserData;
+
+// 🍕 Slice: RTK’s createSlice to create a Redux slice
+export const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    resetUserData: () => {
+      return initialState;
+    },
+  },
```

Comment:
In the docs I sometimes see the getUsers, getUsersById etc. here. Now that we have RTK, are the reducers here only going to be non-CRUD-related things?

Reply:
All CRUD actions are possible with RTK Query endpoints.
Query type endpoints fetch data from the server and mutation type endpoints send data to the server.  A mutation hook, such as 'useLoginMutation', sends data to the "server", in this code base example it sends it to app\api\auth\route.ts, and this payload data can then be used for create, update, or delete operations.